### PR TITLE
CLOUD-429: Support PXC8.0 in operator

### PR DESCRIPTION
**Pull request by blez**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | failed |
| recreate | passed |
| self-healing | passed |
| scaling | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| storage | passed |
| upgrade | passed |
| demand-backup | passed |
| self-healing-advanced | passed |
| limits | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| monitoring | passed |
| one-pod | passed |
| monitoring-2-0 | passed |
| scheduled-backup | passed |
| affinity | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |

File: e2e-tests/functions [hunk 1]
```diff
@@ -273,7 +278,11 @@ get_proxy_primary() {
         ip=$(run_mysql_local 'SELECT hostname FROM runtime_mysql_servers WHERE hostgroup_id=11 AND status="ONLINE";' "$uri" "$pod")
     done
 
-    get_pod_name "$ip"
+    if [[ "$IMAGE_PXC" =~ 8\.0$ ]] ; then
+        echo $ip | cut -d'.' -f1
```
Comment (delgod) [inline e2e-tests/functions]: why we send only the first octet?

File: e2e-tests/limits/conf/no-requests.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ spec:
     image: -pxc
     resources:
       limits:
-        memory: 500M
+        memory: 600M
```
Comment (delgod) [inline e2e-tests/limits/conf/no-requests.yml]: why it is needed?

File: e2e-tests/security-context/run [hunk 3]
```diff
@@ -38,12 +38,16 @@ compare_kubectl job.batch/$cluster-xb-$backup
...
+    pfx="-80"
+fi
+
 desc 'run pvc restore'
 restore="restore-pvc"
 kubectl_bin apply -f "$test_dir/conf/$cluster-$restore.yml"
 wait_backup_restore $restore
-compare_kubectl job.batch/restore-job-$restore-$cluster
-compare_kubectl pod/restore-src-$restore-$cluster
+compare_kubectl job.batch/restore-job-$restore-$cluster $pfx
```
Comment (delgod) [inline e2e-tests/security-context/run]: why pod check was removed?

File: e2e-tests/security-context/run [hunk 4]
```diff
@@ -38,12 +38,16 @@ compare_kubectl job.batch/$cluster-xb-$backup
 desc 'run pvc backup'
 compare_kubectl cronjob/each-hour-pvc
 
+pfx=""
+if [[ "$IMAGE_PXC" =~ 8\.0$ ]]; then
+    pfx="-80"
+fi
+
```
Comment (delgod) [inline e2e-tests/security-context/run]: why it is not part of `compare_kubectl`?

File: e2e-tests/self-healing-advanced/run [hunk 5]
```diff
@@ -4,26 +4,31 @@ set -o errexit
...
+    
+    restartsAfter=$(kubectl get pod self-healing-advanced-pxc-0 -o yaml | grep restartCount: | awk '{print $2}')
 
-    kubectl_bin get pods \
-        | grep $cluster-pxc- \
-        | grep CrashLoopBackOff
+    # check if pod was restarted
+    if [ "$restartsAfter" -eq "$restartsBefore" ] ; then
+        exit 1
+    fi
```
Comment (delgod) [inline e2e-tests/self-healing-advanced/run]: how do we know that became broken?
the previous logic was like - check if pods are definitely crashed.
new logic - just check that Pod have been restarted.

File: e2e-tests/init-deploy/compare/max_allowed_packet-2-80.sql [hunk 6]
```diff
@@ -0,0 +1 @@
+827326464
```
Comment (delgod) [inline e2e-tests/init-deploy/compare/max_allowed_packet-2-80.sql]: I think this file identical to `e2e-tests/init-deploy/compare/max_allowed_packet-2.sql`, so it is not needed.

File: e2e-tests/security-context/compare/job.batch_restore-job-restore-pvc-sec-context-80.yml [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        job-name: restore-job-restore-pvc-sec-context
+    spec:
+      containers:
+      - command:
+        - recovery-pvc-joiner.sh
+        env:
+        - name: RESTORE_SRC_SERVICE
+          value: restore-src-restore-pvc-sec-context
+        - name: XB_USE_MEMORY
+          value: "750000000"
```
Comment (delgod) [inline e2e-tests/security-context/compare/job.batch_restore-job-restore-pvc-sec-context-80.yml]: hm, why operator counts different values for 8.0 and 5.7?

File: e2e-tests/init-deploy/conf/no-proxysql.yml [hunk 8]
```diff
@@ -9,10 +9,10 @@ spec:
     image: -pxc
     configuration: |
       [mysqld]
-      wsrep_debug=ON
+      wsrep_debug=1
```
Comment (delgod) [inline e2e-tests/init-deploy/conf/no-proxysql.yml]: why we need a number (not "ON")?

Comment (blez) [reply e2e-tests/init-deploy/conf/no-proxysql.yml]: PXC8.0 does not understand "ON"

File: e2e-tests/limits/conf/no-requests.yml (hunk 2 above)
Comment (blez) [reply e2e-tests/limits/conf/no-requests.yml]: PXC8.0 needs more memory.

File: e2e-tests/functions (hunk 1 above)
Comment (blez) [reply e2e-tests/functions]: I'ts not octet. It's pod name

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

File: e2e-tests/security-context/run (hunk 3 above)
Comment (blez) [reply e2e-tests/security-context/run]: Probably was removed accidentally https://github.com/percona/percona-xtradb-cluster-operator/pull/337/commits/287bfe4355aa2cf05f114a5d670073d47a120808

File: e2e-tests/init-deploy/compare/max_allowed_packet-2-80.sql (hunk 6 above)
Comment (blez) [reply e2e-tests/init-deploy/compare/max_allowed_packet-2-80.sql]: https://github.com/percona/percona-xtradb-cluster-operator/pull/337/commits/91639c5fde00ad69ef8ad91208ebda0f0f55088d

File: e2e-tests/security-context/run (hunk 4 above)
Comment (blez) [reply e2e-tests/security-context/run]: https://github.com/percona/percona-xtradb-cluster-operator/pull/337/commits/a9f11357452fe547ce7207390a998c985d453cec

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| recreate | failed |
| scaling | failed |
| big-data | failed |
| self-healing | failed |
| demand-backup | passed |
| scaling-proxysql | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| storage | passed |
| limits | passed |
| upgrade | passed |
| one-pod | passed |
| auto-tuning | passed |
| upgrade-consistency | passed |
| monitoring | passed |
| scheduled-backup | failed |
| monitoring-2-0 | passed |
| affinity | passed |
| proxysql-sidecar-res-limits | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| recreate | passed |
| big-data | failed |
| self-healing | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| storage | failed |
| demand-backup | failed |
| upgrade | passed |
| self-healing-advanced | passed |
| scheduled-backup | failed |
| limits | passed |
| upgrade-consistency | passed |
| operator-self-healing | passed |
| monitoring | passed |
| one-pod | passed |
| monitoring-2-0 | passed |
| auto-tuning | passed |
| affinity | passed |
| proxysql-sidecar-res-limits | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| init-deploy | passed |
| recreate | passed |
| big-data | failed |
| self-healing | passed |
| scaling-proxysql | passed |
| storage | passed |
| demand-backup | passed |
| self-healing-advanced | passed |
| upgrade | passed |
| limits | passed |
| monitoring | passed |
| operator-self-healing | passed |
| monitoring-2-0 | passed |
| upgrade-consistency | passed |
| affinity | passed |
| one-pod | passed |
| scheduled-backup | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | failed |
| recreate | passed |
| scaling | passed |
| init-deploy | passed |
| self-healing | failed |
| scaling-proxysql | passed |
| storage | passed |
| self-healing-advanced | passed |
| demand-backup | passed |
| upgrade | passed |
| limits | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| monitoring | passed |
| one-pod | failed |
| auto-tuning | passed |
| monitoring-2-0 | passed |
| scheduled-backup | failed |
| affinity | passed |
| proxysql-sidecar-res-limits | passed |

File: e2e-tests/self-healing-advanced/run (hunk 5 above)
Comment (blez) [reply e2e-tests/self-healing-advanced/run]: https://github.com/percona/percona-xtradb-cluster-operator/pull/337/commits/9c4f04794ac5663d22fcb90f4c41fcea02226fc1

File: e2e-tests/security-context/compare/job.batch_restore-job-restore-pvc-sec-context-80.yml (hunk 7 above)
Comment (blez) [reply e2e-tests/security-context/compare/job.batch_restore-job-restore-pvc-sec-context-80.yml]: https://github.com/percona/percona-xtradb-cluster-operator/pull/337/commits/7846fba987d6e9c36ef1589103c3e4df513068b8
Probably when I run tests I did not have latest version from master branch.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | failed |
| self-healing | passed |
| scaling | passed |
| init-deploy | passed |
| recreate | passed |
| scaling-proxysql | passed |
| storage | passed |
| self-healing-advanced | failed |
| upgrade | passed |
| operator-self-healing | passed |
| limits | passed |
| demand-backup | failed |
| upgrade-consistency | passed |
| one-pod | passed |
| monitoring | passed |
| scheduled-backup | failed |
| auto-tuning | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| proxysql-sidecar-res-limits | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| big-data | failed |
| init-deploy | passed |
| scaling | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| storage | failed |
| self-healing-advanced | failed |
| demand-backup | failed |
| upgrade | passed |
| operator-self-healing | passed |
| limits | passed |
| upgrade-consistency | passed |
| monitoring | passed |
| scheduled-backup | failed |
| one-pod | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |

**Pull request merged**